tests/utils/test_collection_env_guard.py

```
@@ -0,0 +1,116 @@
+
+
+def test_snapshot_collection_env_filters_to_watched_prefixes():
+    env = {
+        "DYNAMO_TEST_FRAMEWORK": "vllm",
+        "DYN_SYSTEM_PORT": "9090",
+        "SGLANG_LOGGING_LEVEL": "debug",
+        "TRTLLM_LOG_LEVEL": "info",
+        "VLLM_NO_USAGE_STATS": "1",
+        "NCCL_DEBUG": "INFO",
+        "CUDA_VISIBLE_DEVICES": "0",
+        "HF_HOME": "/tmp/hf",
```

Ruff (0.15.13)

[error] Probable insecure usage of temporary file or directory: "/tmp/hf"

(S108)